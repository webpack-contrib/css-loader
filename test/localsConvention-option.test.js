import {
  compile,
  getCompiler,
  getErrors,
  getExecutedCode,
  getModuleSource,
  getWarnings,
} from './helpers/index';

describe('"localsConvention" option', () => {
  it('should work when not specified', async () => {
    const compiler = getCompiler(
      './modules/localsConvention/localsConvention.js',
      {
        modules: true,
      }
    );
    const stats = await compile(compiler);

    expect(
      getModuleSource('./modules/localsConvention/localsConvention.css', stats)
    ).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with a value equal to "asIs"', async () => {
    const compiler = getCompiler(
      './modules/localsConvention/localsConvention.js',
      {
        modules: true,
        localsConvention: 'asIs',
      }
    );
    const stats = await compile(compiler);

    expect(
      getModuleSource('./modules/localsConvention/localsConvention.css', stats)
    ).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with a value equal to "camelCase"', async () => {
    const compiler = getCompiler(
      './modules/localsConvention/localsConvention.js',
      {
        modules: true,
        localsConvention: 'camelCase',
      }
    );
    const stats = await compile(compiler);

    expect(
      getModuleSource('./modules/localsConvention/localsConvention.css', stats)
    ).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with a value equal to "camelCaseOnly"', async () => {
    const compiler = getCompiler(
      './modules/localsConvention/localsConvention.js',
      {
        modules: true,
        localsConvention: 'camelCaseOnly',
      }
    );
    const stats = await compile(compiler);

    expect(
      getModuleSource('./modules/localsConvention/localsConvention.css', stats)
    ).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with a value equal to "dashes"', async () => {
    const compiler = getCompiler(
      './modules/localsConvention/localsConvention.js',
      {
        modules: true,
        localsConvention: 'dashes',
      }
    );
    const stats = await compile(compiler);

    expect(
      getModuleSource('./modules/localsConvention/localsConvention.css', stats)
    ).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with a value equal to "dashesOnly"', async () => {
    const compiler = getCompiler(
      './modules/localsConvention/localsConvention.js',
      {
        modules: true,
        localsConvention: 'dashesOnly',
      }
    );
    const stats = await compile(compiler);

    expect(
      getModuleSource('./modules/localsConvention/localsConvention.css', stats)
    ).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
