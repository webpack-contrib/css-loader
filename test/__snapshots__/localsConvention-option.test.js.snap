// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"localsConvention" option should work when not specified: errors 1`] = `Array []`;

exports[`"localsConvention" option should work when not specified: module 1`] = `
"// Imports
var ___CSS_LOADER_API_IMPORT___ = require(\\"../../../../src/runtime/api.js\\");
exports = ___CSS_LOADER_API_IMPORT___(false);
// Module
exports.push([module.id, \\".erBXHZCN_thRYfCnk-aH8 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._2YsQE-S0o0NRXfC6XNApz2 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._3gGBcJHZU3seQVP5aq7Ksq {\\\\n  color: red;\\\\n}\\\\n\\\\na {\\\\n  color: yellow;\\\\n}\\\\n\\", \\"\\"]);
// Exports
exports.locals = {
	\\"foo\\": \\"bar\\",
	\\"my-btn-info_is-disabled\\": \\"value\\",
	\\"btn-info_is-disabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btn--info_is-disabled_1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"simple\\": \\"_3gGBcJHZU3seQVP5aq7Ksq\\"
};
module.exports = exports;
"
`;

exports[`"localsConvention" option should work when not specified: result 1`] = `
Array [
  Array [
    "./modules/localsConvention/localsConvention.css",
    ".erBXHZCN_thRYfCnk-aH8 {
  color: blue;
}

._2YsQE-S0o0NRXfC6XNApz2 {
  color: blue;
}

._3gGBcJHZU3seQVP5aq7Ksq {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`"localsConvention" option should work when not specified: warnings 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "asIs": errors 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "asIs": module 1`] = `
"// Imports
var ___CSS_LOADER_API_IMPORT___ = require(\\"../../../../src/runtime/api.js\\");
exports = ___CSS_LOADER_API_IMPORT___(false);
// Module
exports.push([module.id, \\".erBXHZCN_thRYfCnk-aH8 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._2YsQE-S0o0NRXfC6XNApz2 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._3gGBcJHZU3seQVP5aq7Ksq {\\\\n  color: red;\\\\n}\\\\n\\\\na {\\\\n  color: yellow;\\\\n}\\\\n\\", \\"\\"]);
// Exports
exports.locals = {
	\\"foo\\": \\"bar\\",
	\\"my-btn-info_is-disabled\\": \\"value\\",
	\\"btn-info_is-disabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btn--info_is-disabled_1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"simple\\": \\"_3gGBcJHZU3seQVP5aq7Ksq\\"
};
module.exports = exports;
"
`;

exports[`"localsConvention" option should work with a value equal to "asIs": result 1`] = `
Array [
  Array [
    "./modules/localsConvention/localsConvention.css",
    ".erBXHZCN_thRYfCnk-aH8 {
  color: blue;
}

._2YsQE-S0o0NRXfC6XNApz2 {
  color: blue;
}

._3gGBcJHZU3seQVP5aq7Ksq {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`"localsConvention" option should work with a value equal to "asIs": warnings 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "camelCase": errors 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "camelCase": module 1`] = `
"// Imports
var ___CSS_LOADER_API_IMPORT___ = require(\\"../../../../src/runtime/api.js\\");
exports = ___CSS_LOADER_API_IMPORT___(false);
// Module
exports.push([module.id, \\".erBXHZCN_thRYfCnk-aH8 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._2YsQE-S0o0NRXfC6XNApz2 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._3gGBcJHZU3seQVP5aq7Ksq {\\\\n  color: red;\\\\n}\\\\n\\\\na {\\\\n  color: yellow;\\\\n}\\\\n\\", \\"\\"]);
// Exports
exports.locals = {
	\\"foo\\": \\"bar\\",
	\\"my-btn-info_is-disabled\\": \\"value\\",
	\\"myBtnInfoIsDisabled\\": \\"value\\",
	\\"btn-info_is-disabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btnInfoIsDisabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btn--info_is-disabled_1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"btnInfoIsDisabled1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"simple\\": \\"_3gGBcJHZU3seQVP5aq7Ksq\\"
};
module.exports = exports;
"
`;

exports[`"localsConvention" option should work with a value equal to "camelCase": result 1`] = `
Array [
  Array [
    "./modules/localsConvention/localsConvention.css",
    ".erBXHZCN_thRYfCnk-aH8 {
  color: blue;
}

._2YsQE-S0o0NRXfC6XNApz2 {
  color: blue;
}

._3gGBcJHZU3seQVP5aq7Ksq {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`"localsConvention" option should work with a value equal to "camelCase": warnings 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "camelCaseOnly": errors 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "camelCaseOnly": module 1`] = `
"// Imports
var ___CSS_LOADER_API_IMPORT___ = require(\\"../../../../src/runtime/api.js\\");
exports = ___CSS_LOADER_API_IMPORT___(false);
// Module
exports.push([module.id, \\".erBXHZCN_thRYfCnk-aH8 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._2YsQE-S0o0NRXfC6XNApz2 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._3gGBcJHZU3seQVP5aq7Ksq {\\\\n  color: red;\\\\n}\\\\n\\\\na {\\\\n  color: yellow;\\\\n}\\\\n\\", \\"\\"]);
// Exports
exports.locals = {
	\\"foo\\": \\"bar\\",
	\\"myBtnInfoIsDisabled\\": \\"value\\",
	\\"btnInfoIsDisabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btnInfoIsDisabled1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"simple\\": \\"_3gGBcJHZU3seQVP5aq7Ksq\\"
};
module.exports = exports;
"
`;

exports[`"localsConvention" option should work with a value equal to "camelCaseOnly": result 1`] = `
Array [
  Array [
    "./modules/localsConvention/localsConvention.css",
    ".erBXHZCN_thRYfCnk-aH8 {
  color: blue;
}

._2YsQE-S0o0NRXfC6XNApz2 {
  color: blue;
}

._3gGBcJHZU3seQVP5aq7Ksq {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`"localsConvention" option should work with a value equal to "camelCaseOnly": warnings 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "dashes": errors 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "dashes": module 1`] = `
"// Imports
var ___CSS_LOADER_API_IMPORT___ = require(\\"../../../../src/runtime/api.js\\");
exports = ___CSS_LOADER_API_IMPORT___(false);
// Module
exports.push([module.id, \\".erBXHZCN_thRYfCnk-aH8 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._2YsQE-S0o0NRXfC6XNApz2 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._3gGBcJHZU3seQVP5aq7Ksq {\\\\n  color: red;\\\\n}\\\\n\\\\na {\\\\n  color: yellow;\\\\n}\\\\n\\", \\"\\"]);
// Exports
exports.locals = {
	\\"foo\\": \\"bar\\",
	\\"my-btn-info_is-disabled\\": \\"value\\",
	\\"myBtnInfo_isDisabled\\": \\"value\\",
	\\"btn-info_is-disabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btnInfo_isDisabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btn--info_is-disabled_1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"btnInfo_isDisabled_1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"simple\\": \\"_3gGBcJHZU3seQVP5aq7Ksq\\"
};
module.exports = exports;
"
`;

exports[`"localsConvention" option should work with a value equal to "dashes": result 1`] = `
Array [
  Array [
    "./modules/localsConvention/localsConvention.css",
    ".erBXHZCN_thRYfCnk-aH8 {
  color: blue;
}

._2YsQE-S0o0NRXfC6XNApz2 {
  color: blue;
}

._3gGBcJHZU3seQVP5aq7Ksq {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`"localsConvention" option should work with a value equal to "dashes": warnings 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "dashesOnly": errors 1`] = `Array []`;

exports[`"localsConvention" option should work with a value equal to "dashesOnly": module 1`] = `
"// Imports
var ___CSS_LOADER_API_IMPORT___ = require(\\"../../../../src/runtime/api.js\\");
exports = ___CSS_LOADER_API_IMPORT___(false);
// Module
exports.push([module.id, \\".erBXHZCN_thRYfCnk-aH8 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._2YsQE-S0o0NRXfC6XNApz2 {\\\\n  color: blue;\\\\n}\\\\n\\\\n._3gGBcJHZU3seQVP5aq7Ksq {\\\\n  color: red;\\\\n}\\\\n\\\\na {\\\\n  color: yellow;\\\\n}\\\\n\\", \\"\\"]);
// Exports
exports.locals = {
	\\"foo\\": \\"bar\\",
	\\"myBtnInfo_isDisabled\\": \\"value\\",
	\\"btnInfo_isDisabled\\": \\"erBXHZCN_thRYfCnk-aH8\\",
	\\"btnInfo_isDisabled_1\\": \\"_2YsQE-S0o0NRXfC6XNApz2\\",
	\\"simple\\": \\"_3gGBcJHZU3seQVP5aq7Ksq\\"
};
module.exports = exports;
"
`;

exports[`"localsConvention" option should work with a value equal to "dashesOnly": result 1`] = `
Array [
  Array [
    "./modules/localsConvention/localsConvention.css",
    ".erBXHZCN_thRYfCnk-aH8 {
  color: blue;
}

._2YsQE-S0o0NRXfC6XNApz2 {
  color: blue;
}

._3gGBcJHZU3seQVP5aq7Ksq {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`"localsConvention" option should work with a value equal to "dashesOnly": warnings 1`] = `Array []`;
